ESMF/UFS/Aerosol_Comp_Mod.F90

```diff
@@ -709,6 +720,101 @@ subroutine AerosolTracerUpdate(state, cap, rc)
 
   end subroutine AerosolTracerUpdate
 
+  subroutine AerosolDiagUpdate(model, cap, rc)
+    type(ESMF_GridComp)            :: model
+    type(MAPL_Cap)                 :: cap
+    integer, optional, intent(out) :: rc
+
+    ! -- local variables
+    integer :: localrc
+    integer :: i, j, k, kk, ni, nj, nk
+    real(ESMF_KIND_R4) :: pm25, pm10
+    real(ESMF_KIND_R4), dimension(:,:,:),   pointer :: rho
+    real(ESMF_KIND_R8), dimension(:,:,:,:), pointer :: q
+
+    type(ESMF_Field) :: field
+    type(ESMF_State) :: state
+
+    ! -- local parameters
+    real(ESMF_KIND_R8), parameter :: w_du2  = log(1.250_ESMF_KIND_R8) / log(1.8_ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_du4  = log(1.667_ESMF_KIND_R8) / log(2.0_ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_ss3  = log(2.50_ESMF_KIND_R8) / log(3._ESMF_KIND_R8)
+    real(ESMF_KIND_R8), parameter :: w_so4  = 132.14_ESMF_KIND_R8 / 96.06_ESMF_KIND_R8
+    real(ESMF_KIND_R8), parameter :: w_no3  = 80.043_ESMF_KIND_R8 / 62.0_ESMF_KIND_R8
+    real(ESMF_KIND_R8), parameter :: w25_no3an2 = 0.138_ESMF_KIND_R8 * w_no3
+    real(ESMF_KIND_R8), parameter :: w10_no3an2 = 0.808_ESMF_KIND_R8 * w_no3
+    real(ESMF_KIND_R8), parameter :: w10_no3an3 = 0.164_ESMF_KIND_R8 * w_no3
```

Comment:
Is r8 really necessary?

Reply:
@adarmenov - no. However, PM2.5 and PM10 are computed using tracer mixing ratios stored in an `ESMF_TYPEKIND_R8` coupling field, so `ESMF_KIND_R8` is being used for consistency.